elasticdl/python/master/task_dispatcher.py

```diff
@@ -134,6 +143,41 @@ def create_tasks(self, task_type, model_version=-1):
                 self._todo.extend(tasks)
         return tasks
 
+    def _create_save_model_task(self):
+        # For the SavedModel Task, we need a shard of data to build the model
+        shards = self._training_shards
+        assert shards is not None
+
+        (shard_name, (start_ind_this_shard, num_records_this_shard)) = next(
+            iter(shards.items())
+        )
+        start_ind_this_task = start_ind_this_shard
+        end_ind_this_task = start_ind_this_shard + min(
+            self._records_per_task, num_records_this_shard
+        )
+
+        # Use the first shard of data to do the SavedModel work
+        save_model_task = _Task(
+            shard_name=shard_name,
+            start=start_ind_this_task,
+            end=end_ind_this_task,
+            type=elasticdl_pb2.SAVE_MODEL,
+        )
+
+        self._todo.append(save_model_task)
```

Comment:
The `save_model_task` should be invoked at the end of the job, right? But if we insert `save_model_task` into `_todo` at the beginning, it means the task will be executed before training? Do I miss anything here?

Reply:
Yes, it will be executed before training if the task is inserted at the head before the training starts.
So in this PR, we add a callback to create save model task in _task_list_done_callbacks. The callback will only be invoked when all the training/evaluation/prediction task complete